include/openssl/core.h

```diff
@@ -82,35 +82,33 @@ struct ossl_param_st {
 
 /* Currently supported OSSL_PARAM data types */
 /*
- * OSSL_PARAM_INTEGER and OSSL_PARAM_UNSIGNED_INTEGER
- * are arbitrary length and therefore require an arbitrarily sized buffer,
- * since they may be used to pass numbers larger than what is natively
- * available.
- *
- * The number must be buffered in native form, i.e. MSB first on B_ENDIAN
- * systems and LSB first on L_ENDIAN systems.  This means that arbitrary
- * native integers can be stored in the buffer, just make sure that the
- * buffer size is correct and the buffer itself is properly aligned (for
- * example by having the buffer field point at a C integer).
+ * Numbers are stored in native form.
  */
-# define OSSL_PARAM_INTEGER              1
-# define OSSL_PARAM_UNSIGNED_INTEGER     2
-/*-
- * OSSL_PARAM_REAL
- * is a C binary floating point values in native form and alignment.
+# define OSSL_PARAM_INT                 1
+# define OSSL_PARAM_UINT                2
+# define OSSL_PARAM_INT64               3
+# define OSSL_PARAM_UINT64              4
+# define OSSL_PARAM_LONG                5
+# define OSSL_PARAM_ULONG               6
+# define OSSL_PARAM_SIZET               7
+# define OSSL_PARAM_DOUBLE              8
+
+/*- OSSL_PARAM_BIGNUM
+ * An OpenSSL BIGNUM; stored in native big/little endian format.
  */
-# define OSSL_PARAM_REAL                 3
+# define OSSL_PARAM_BIGNUM              0x20
```

Comment:
Why the sudden switch to hex?

Reply:
I wondered the same thing.

Reply:
I'm sorry, I did not want to mark this item resolved. I was just scrolling through the comments on my mobile when my thumb happened to hit the button 😊.

Reply:
Because I added the numbering gap later.  Converted all to hex and added a comment.

Reply:
Not sure I understand why hex is used. "coolness factor"?

Reply:
To make it very obvious how big the gap for "unused integral types" is.

Reply:
Because somehow, decimal math (32 - 8) is harder than hexadecimal math (0x20 - 0x08)?